libs/libskynet/src-test/encrypt.test.ts

```diff
@@ -0,0 +1,35 @@
+import { bufToHex } from "../src/encoding.js";
+import { otpEncrypt } from "../src/encrypt.js";
+import { sha512 } from "../src/sha512.js";
+
+test("otpEncrypt", () => {
+  // Perform a basic encryption and ensure that the data changes.
+  let initialData1 = new TextEncoder().encode("this is a test string to encrypt");
+  let initialData2 = new TextEncoder().encode("this is a test string to encrypt");
+  let key1 = sha512(new TextEncoder().encode("this is a key preimage"));
+  let key2 = sha512(new TextEncoder().encode("this is a different key preimage"));
+  console.log("before encrypt:", bufToHex(initialData2));
+  otpEncrypt(key1, initialData2);
+  console.log("after encrypt: ", bufToHex(initialData2));
+  expect(initialData1.length).toBe(initialData2.length);
+  expect(initialData1).not.toEqual(initialData2);
+
+  // Check that decryption works.
+  otpEncrypt(key1, initialData2);
+  expect(initialData1).toEqual(initialData2);
+
+  // Check that encrypting with a different key will give a different data.
```

Comment:
I don't see the value of splitting them up, it'll result in more test code and repeated setup.

Reply:
> I don't see the value of splitting them up

Splitting the code into smaller pieces improves readability and ensures that each functionality and use case is tested. Smaller, self descriptive tests document the code and when they fail, it's immediately obvious which test case is broken. In jest, tests fail early on first failed assertion so running a long block might fail on first assertion testing a specific use case and you wouldn't know if the issue is limited to that use case only or it would fail on other use cases too.

> it'll result in more test code and repeated setup

More test code is actually a good thing since it's also a documentation and setup time in jest is negligible

Reply:
You are talking about breaking up a test that is already only 21 lines of code. There's also a tradeoff with how long the test takes to write, I really don't think asking someone to break up a test that's  already only 21 lines of code (including comments and whitespace) is a good use of time and code iteration. There's a lot to get done.

Reply:
You are talking to me like I've never written a test suite before. This isn't intended to be a set of unit tests, it's smoke testing to verify that a couple of end-to-end behaviors match the expectation. In practice I've found smoke testing to be significantly more reliable at exposing bugs than isolated unit tests because isolated unit tests frequently miss emergent behaviors when multiple pieces of code are interacting fully.

I understand the value of unit testing, but we wouldn't even be having this discussion if I hadn't written the test in the first place - you haven't requested unit tests for many other pieces of code that are missing them. This test is plenty simple that if something goes wrong, someone can jump in, read through the steps, and start debugging.